### PR TITLE
Move rich binding implementations from the worker

### PR DESCRIPTION
**Pull request by elprans**
Per Azure/azure-functions-python-worker#409, the rich binding
implementations are now in the library.

### CODE REVIEW

File: azure/functions/_thirdparty/typing_inspect.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+def _gorg(cls):
+    """This function exists for compatibility with old typing versions."""
+    assert isinstance(cls, GenericMeta)
+    if hasattr(cls, '_gorg'):
+        return cls._gorg
+    while cls.__origin__ is not None:
+        cls = cls.__origin__
+    return cls
+
+
```
Comment (pragnagopa) [inline azure/functions/_thirdparty/typing_inspect.py]: This looks like a general utility. Shoudn't it be part of the python worker?

File: azure/functions/__init__.py [hunk 2]
```diff
@@ -6,8 +6,23 @@
...
+from . import eventhub  # NoQA
+from . import http  # NoQA
+from . import queue  # NoQA
+from . import servicebus  # NoQA
+from . import timer  # NoQA
+
 
 __all__ = (
+    # Functions
+    'get_registry',
```
Comment (pragnagopa) [inline azure/functions/__init__.py]: Rename to be more explicit. TriggerMetadataRegistry?

File: azure/functions/_thirdparty/typing_inspect.py (hunk 1 above)
Comment (elprans) [reply azure/functions/_thirdparty/typing_inspect.py]: It already is, but I'd like to avoid adding a dependency on the worker.

**Comment by elprans** [comment]
>  Will you be porting unit tests as well?

We need to figure out the dependency story first.  Tests depend on the worker.

**Comment by pragnagopa** [comment]
> We need to figure out the dependency story first. Tests depend on the worker.

E2E tests depend on worker. Unit tests should be moved here. Separate PR is ok.

**Comment by maiqbal11** [comment]
@elprans, we'll need to do a release with this change in order to run the tests for https://github.com/Azure/azure-functions-python-worker/pull/409 in the CI?

**Comment by elprans** [comment]
@maiqbal11 Yes, we need to merge this and do a release.

**Comment by maiqbal11** [comment]
> @maiqbal11 Yes, we need to merge this and do a release.

Are we blocked on performing the merge for this? Would be good to get this in so we can start seeing CI builds for the generic bindings PR.

**Comment by elprans** [comment]
@maiqbal11 Published azure-functions-python-library-1.0.0b5

**Pull request merged**